change/@ni-nimble-components-30319fbf-4709-43b2-9ab5-378704cc1cff.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
I love small PRs but a couple things become more important when slicing up work like this:
1. providing more big-picture context about how this will be used. Do we have a comment / HLD / draft PR explaining what the current proposal is to fix the router bug? How can I gain confidence that these APIs are needed?
2. providing more detail about the testing plan. There aren't any auto tests of these new APIs in this PR. You mentioned that you tested them in another branch but are those auto tests or one-time-prove-it-works manual tests? If auto tests, why are we relying on integration tests in nimble-angular rather than unit tests in nimble-components? There could be good reasons, but the context helps understand those reasons.

Reply:
1. There's no HLD or such document, but I do have a draft PR for the follow-on bug fix: #1349. This approach was discussed over a couple of conversations between Molly, Milan, and myself. Before starting down this path, I got consensus from them that this was how we thought it should be addressed.
2. I did one-time manual testing in my prototype build of TestInsights, but I also have a couple automated tests in my next PR (in `nimble-angular`). One of those tests asserts that the row id gets passed to the callback function, thereby validating the changes in this PR. I didn't think it necessary to test that the `rowRecordId` properties I added get populated correctly, because those are essentially implementation details. As far as validating that the row id ends up in the delegated event details, we do have a test in `table-cell-view.spec` that verifies the behavior of delegated events, but it doesn't do it in the context of a table with row data (so there are no row ids to test). I figured it was already going to be tested by my `nimble-angular` test, so I didn't bother to write a new test in `nimble-components`.

Reply:
>  I figured it was already going to be tested by my nimble-angular test, so I didn't bother to write a new test in nimble-components.

This is a feature of the delegated events implementation so it should have it's own test coverage.